### DISCUSSION TITLE
Why does `get_real`, `get_int` etc. pack before unpacking?

### DISCUSSION THREAD

**Discussion by GaffaSnobb** [gijzelaerr/python-snap7#532] [Q&A] [answered]
Hi there, so a quick question. The util [get_real](https://github.com/gijzelaerr/python-snap7/blob/e0aaed2c3109b372e853da8835cf6599698ad924/snap7/util/getters.py#L124) calls `struct.pack` before unpacking with `struct.unpack`. What is the reason for this?

Thanks : )

**Comment by gijzelaerr** [comment, marked as answer]
Closing as stale. The pack-before-unpack pattern converts individual bytes into the correct binary representation before interpreting them. It is a common idiom with `struct`.
